src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1249,39 +1249,76 @@ private void ImportThrow()
         {
         }
 
+        private LLVMValueRef GetInstanceFieldAddress(StackEntry objectEntry, FieldDesc field)
+        {
+            var objectType = objectEntry.Type ?? field.OwningType;
+            LLVMValueRef typedObjectValue;
+            if (objectType.IsValueType && !objectType.IsPointer && objectEntry.Kind != StackValueKind.NativeInt)
+            {
+                typedObjectValue = LLVM.BuildAlloca(_builder, GetLLVMTypeForTypeDesc(objectType), "objptr");
+                LLVM.BuildStore(_builder, objectEntry.LLVMValue, typedObjectValue);
+            }
+            else
+            {
+                typedObjectValue = objectEntry.LLVMValue;
+            }
+
+            var untypedObjectPointer = LLVM.BuildPointerCast(_builder, typedObjectValue, LLVM.PointerType(LLVMTypeRef.Int8Type(), 0), String.Empty);
+            var loadLocation = LLVM.BuildGEP(_builder, untypedObjectPointer,
+            new LLVMValueRef[] { LLVM.ConstInt(LLVM.Int32Type(), (ulong)field.Offset.AsInt, LLVMMisc.False) }, String.Empty);
+            return LLVM.BuildPointerCast(_builder, loadLocation, LLVM.PointerType(GetLLVMTypeForTypeDesc(field.FieldType), 0), String.Empty);
+        }
+
+        private LLVMValueRef GetFieldAddress(FieldDesc field, bool isStatic)
+        {
+            if (field.IsStatic)
+            {
+                //pop unused value
+                if (!isStatic)
+                    _stack.Pop();
+
+                LLVMValueRef untypedFieldAddress = WebAssemblyObjectWriter.EmitGlobal(Module, field, _compilation.NameMangler);
+                return LLVM.BuildPointerCast(_builder, untypedFieldAddress, LLVM.PointerType(GetLLVMTypeForTypeDesc(field.FieldType), 0), string.Empty);
+            }
+            else
+            {
+                return GetInstanceFieldAddress(_stack.Pop(), field);
+            }
+        }
+
         private void ImportLoadField(int token, bool isStatic)
         {
+            FieldDesc field = (FieldDesc)_methodIL.GetObject(token);
+            LLVMValueRef fieldAddress = GetFieldAddress(field, isStatic);
+            LLVMValueRef loadValue = LLVM.BuildLoad(_builder, fieldAddress, "ldfld_" + field.Name);
+            PushExpression(GetStackValueKind(field.FieldType), "ldfld", loadValue, field.FieldType);
         }
 
         private void ImportAddressOfField(int token, bool isStatic)
         {
+            FieldDesc field = (FieldDesc)_methodIL.GetObject(token);
+            LLVMValueRef fieldAddress = GetFieldAddress(field, isStatic);
+            PushExpression(StackValueKind.NativeInt, "ldflda", fieldAddress, field.FieldType.MakePointerType());
```

Comment:
I'm not certain on this, but the ILToCppImporter suggests it's a ByRef, not a NativeInt